packages/graphql/src/translate/connection-clause/get-sort-fields.ts

```diff
@@ -18,25 +18,21 @@
  */
 
 import type { ResolveTree } from "graphql-parse-resolve-info";
-import type { ConnectionSortArg, GraphQLSortArg } from "../../types";
+import type { GraphQLSortArg } from "../../types";
 
 type SortFields = {
-    edge: GraphQLSortArg;
-    node: GraphQLSortArg;
+    edge?: GraphQLSortArg;
+    node?: GraphQLSortArg;
 };
 
-/** Returns sort fields for connections, split between edge and node */
-export function getSortFields(resolveTree: ResolveTree): SortFields {
-    const sortInput = (resolveTree.args.sort ?? []) as ConnectionSortArg[];
-
-    return {
-        edge: getSortFieldsByElement(sortInput, "edge"),
-        node: getSortFieldsByElement(sortInput, "node"),
-    };
+/** Returns sort fields for connections, of format [{ ['edge' | 'node']: { [key]: [value] } }] */
```

Comment:
yes that's correct, what I wanted to highlight is that either an `edge` or a `node` is present, but not both

Reply:
Maybe we could improve typings to reflect that?

```ts
type SortFields = { edge: GraphQLSortArg } | { node: GraphQLSortArg }
```

Again, just a non-blocking nitpick, in case you think is worth it

Reply:
sounds good